toolchain/check/deduce.cpp

```diff
@@ -503,6 +503,25 @@ auto DeductionContext::Deduce() -> bool {
   return true;
 }
 
+// Gets the entity name of a generic binding. The generic binding may be an
+// imported instruction.
+static auto GetEntityNameForGenericBinding(Context& context,
+                                           SemIR::InstId binding_id)
+    -> SemIR::NameId {
+  // If `binding_id` is imported, it may not have an entity name. Get a
+  // canonical local instruction from its constant value which does.
+  if (context.insts().Is<SemIR::ImportRefLoaded>(binding_id)) {
+    binding_id = context.constant_values().GetConstantInstId(binding_id);
+  }
```

Comment:
Consider calling `GetConstantInstId` unconditionally rather than coupling this code to `ImportRefLoaded`. That would also handle other ways the binding ID might be being referenced indirectly (there are none so far, I think, but there could be others later).

Reply:
Ok, thanks, done https://github.com/carbon-language/carbon-lang/pull/5034